roomacoustics/sti.py

```diff
@@ -0,0 +1,176 @@
+    11200])[1:-1,:])
+# MTF per octave and modulation frequency
+    mf = [0.63, 0.80, 1, 1.25, 1.60, 2, 2.5, 3.15, 4, 5, 6.3, 8, 10, 12]
+    mtf = np.zeros((14,)+sig_oct.cshape)
+    sig_en = np.sum(np.square(sig_oct.time),axis=-1)
+    sig_oct_newlength = sig_oct.copy()
+
+    #calculate mtf
+    with np.errstate(divide='ignore'):  #return nan for empty IR
+        for i in range(len(mf)):
+            fbin = int(np.ceil(mf[i]*sig_oct.n_samples/sig_oct.sampling_rate))
+            newlength = int(sig_oct.sampling_rate*fbin/mf[i])
+            sig_oct_newlength.time = np.concatenate((sig_oct.time, 
+                np.zeros((sig_oct.cshape + (newlength-sig_oct.n_samples,)))),
+                axis=-1)
+            sig_oct_newlength_sq = Signal(np.square(sig_oct_newlength.time),
+                sig_oct.sampling_rate)
+            mtf[i]=abs(sig_oct_newlength_sq.freq[...,fbin])/sig_en
+
+
+# Adjustment of mtf for ambient noise, auditory masking and threshold effects
+
+    # calculation only if levels are given
+    if not np.isnan(level).all():
+        # warn if SNR is below 20 dB in any band
+        level = level
+        snr = snr
+        if np.any(snr<20):
+            warnings.warn("SNR should be at least 20 dB for every octave band.")
+        # set snr to infinity if not given
+        if np.isnan(snr).all():
+            warnings.warn("SNR is not or incompletely given, calculation with \
+            infinite SNR.")
+            snr = np.ones(level.shape)*np.inf
+        # overall intensity
+        i_k = 10**(level/10)+10**((level-snr)/10)
+        # Apply ambient noise effects
+        mtf =  mtf*(10**(level.T/10)/i_k.T)
+
+        # consideration of auditory effects only for acoustical signals
+        if data_type == "electrical":
+            pass
+        else: 
+            if data_type != "acoustical":
+                warnings.warn("Data type is considered as acoustical. \
+                Consideration of masking effects not valid for electrically \
+                obtained signals.") 
+            
+            # level-dependent auditory masking
+            amdb = level.copy()
+            amdb[amdb<63]=0.5*amdb[amdb<63]-65
+            amdb[(63<=amdb) & (amdb<67)]=1.8*amdb[(63<=amdb) & (amdb<67)]-146.9
+            amdb[(67<=amdb) & (amdb<100)]=0.5*amdb[(67<=amdb) & (amdb<100)]-59.8
+            amdb[100<=amdb]=amdb[100<=amdb]-10
+            amf = 10**(amdb/10)
+
+            # masking intensity
+            i_am = np.zeros(i_k.shape)
+            i_am = i_k*amf
+
+            # absolute speech reception threshold
+            artdb = np.array([[46,27,12,6.5,7.5,8,12]])
+            i_rt = 10**(artdb/10)
+            # apply auditory and masking effects
+            i_T = i_k/(i_k+i_am+i_rt)
+            i_T = np.squeeze(i_T)
+            mtf = mtf*i_T.T
+
+    else:
+       warnings.warn("Signal level not or incompletely given. Auditory and \
+           masking effects not considered.")
```

Comment:
As commented above, I would raise a ValueError in this case. Otherwise we would do things that are not expected by the user.